Dockerfile

```
@@ -1,14 +1,22 @@
-FROM golang:1.17.5-alpine3.15 as builder
-RUN apk add --no-cache git ca-certificates make tzdata
-COPY . /app
-RUN cd /app && \
-    go get -d -v && \
-    CGO_ENABLED=0 GOOS=linux go build -v -a -installsuffix cgo -o prometheus_bot
+FROM golang:1.17.6-alpine3.15 as builder
+RUN addgroup -S appgroup && adduser -S appuser -G appgroup -u 1000 \
+  && apk add --update --no-cache ca-certificates tzdata
+WORKDIR /app
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
+COPY . .
```

What do you think about moving it before `RUN go mod download` and delete two lines copying `go.mod` and `go.sum`?
That will be easier to read and understand later, is there a particular reason you're doing it this way?
```suggestion
COPY . .
```

---

Yep, because of docker cache.
Just try to change something in the code and run rebuild.
You will see donwloading of go modules complitely skiping because this layer is chached.

---

This only helps in `go.mod` and `go.sum` change, I don't know how much that changes in development, let's see what the maintainer thinks though.